cli/commands/play.go

```diff
@@ -522,34 +524,77 @@ func buildSnakesFromOptions() map[string]SnakeState {
 func printMap(state *rules.BoardState, snakeStates map[string]SnakeState, gameTurn int32) {
 	var o bytes.Buffer
 	o.WriteString(fmt.Sprintf("Ruleset: %s, Seed: %d, Turn: %v\n", GameType, Seed, gameTurn))
-	board := make([][]rune, state.Width)
+	board := make([][]string, state.Width)
 	for i := range board {
-		board[i] = make([]rune, state.Height)
+		board[i] = make([]string, state.Height)
 	}
 	for y := int32(0); y < state.Height; y++ {
 		for x := int32(0); x < state.Width; x++ {
-			board[x][y] = '◦'
+			if UseColor {
+				board[x][y] = "\033[38;2;200;200;200m□"
+			} else {
+				board[x][y] = "◦"
+			}
 		}
 	}
 	for _, oob := range state.Hazards {
-		board[oob.X][oob.Y] = '░'
+		if UseColor {
+			board[oob.X][oob.Y] = "\033[48;2;127;127;127m \033[107m"
+		} else {
+			board[oob.X][oob.Y] = "░"
+		}
+	}
+	if UseColor {
+		o.WriteString(fmt.Sprintf("Hazards \033[48;2;127;127;127m \033[0m: %v\n", state.Hazards))
+	} else {
+		o.WriteString(fmt.Sprintf("Hazards ░: %v\n", state.Hazards))
 	}
-	o.WriteString(fmt.Sprintf("Hazards ░: %v\n", state.Hazards))
 	for _, f := range state.Food {
-		board[f.X][f.Y] = '⚕'
+		if UseColor {
+			board[f.X][f.Y] = "\033[38;2;255;92;117m●"
+		} else {
+			board[f.X][f.Y] = "⚕"
+		}
+	}
+	if UseColor {
+		o.WriteString(fmt.Sprintf("Food \033[38;2;255;92;117m\033[107m●\033[0m: %v\n", state.Food))
+	} else {
+		o.WriteString(fmt.Sprintf("Food ⚕: %v\n", state.Food))
 	}
-	o.WriteString(fmt.Sprintf("Food ⚕: %v\n", state.Food))
 	for _, s := range state.Snakes {
+		var red, green, blue int64 = 136, 136, 136
+		if len(snakeStates[s.ID].Color) == 7 {
+			snake_red, err_r := strconv.ParseInt(snakeStates[s.ID].Color[1:3], 16, 64)
+			snake_green, err_g := strconv.ParseInt(snakeStates[s.ID].Color[3:5], 16, 64)
+			snake_blue, err_b := strconv.ParseInt(snakeStates[s.ID].Color[5:], 16, 64)
+			if err_r == nil && err_g == nil && err_b == nil {
+				red, green, blue = snake_red, snake_green, snake_blue
+			}
+		}
 		for _, b := range s.Body {
 			if b.X >= 0 && b.X < state.Width && b.Y >= 0 && b.Y < state.Height {
-				board[b.X][b.Y] = snakeStates[s.ID].Character
+				if UseColor {
+					board[b.X][b.Y] = fmt.Sprintf("\033[38;2;%d;%d;%dm■", red, green, blue)
```

Comment:
It looks like in royale games, snakes and food over hazards make the hazards invisible. Do you think it's doable to apply the hazard background afterwards in this case, so it still shows up?

Reply:
I don't think that's possible without i rewrite, maybe something for a different PR